File: cpp/max_repeat_seq.cpp
#include <iostream>

int main(int argc, char *argv[])
{

    if (argc == 2 && strcmp(argv[1], "--help") == 0) {
        std::cout << "Set sequences as args after calling app; example: 111100 000110110 etc" << std::endl;
        return 0;
    }

    for (int i = 1; i < argc; ++i) {
        int line_unit = 0;
        int line_unit_counter = 0;
        for (int j = 0; argv[i][j] != '\0'; j++) {
            int counter = argv[i][j] - '0';
            if (counter == 1) {
                line_unit_counter++;
                line_unit = (line_unit < line_unit_counter) ? line_unit_counter : line_unit;
            } else {
                line_unit_counter = 0;
            }
        }
        std::cout << "max sequences range is: " << line_unit << std::endl;
    }

    return 0;
}

File: php/max_repeat_seq.php
<?php
$arr = [0, 0, 1, 0, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1,];

$options = getopt("h::s::", array("help::"));
$helpArr = ['h'=>'try to use [-s] params for setting sequences 0 and 1; example: -s111000'];
//print_r($options);
if(isset($options['h']) || isset($options['help']))
{
    echo array_merge($options, $helpArr)['h'] . PHP_EOL;
    exit(0);
}

if(isset($argv[1]))
{
    $arr = str_split(trim($argv[1]));
}
if(isset($options['s']))
{
    $arr = str_split(trim($options['s']));
}

$line_unit = 0;
$line_unit_counter = 0;
foreach ($arr as $key => $val) {
    if ($val == 1) {
        $line_unit_counter++;
        $line_unit = ($line_unit < $line_unit_counter) ? $line_unit_counter : $line_unit;
    } else {
        $line_unit_counter = 0;
    }
}

echo $line_unit . PHP_EOL;